app/(dashboard)/notes/page.tsx

```diff
@@ -0,0 +1,79 @@
+import NotesTable, { NoteTableView } from '@/components/NotesTable';
+import { createClient } from '@/utils/supabase/server';
+import Link from 'next/link';
+import { redirect } from 'next/navigation';
+import { PiCaretLeft } from 'react-icons/pi';
+
+export default async function Notes({
+  searchParams,
+}: {
+  searchParams: { search?: string; tag?: string };
+}) {
+  const supabase = createClient();
+
+  const {
+    data: { user },
+  } = await supabase.auth.getUser();
+
+  if (!user) {
+    return redirect('/login');
+  }
+
+  const { search, tag } = searchParams;
+
+  if (!search && !tag) {
+    return redirect('/');
+  }
+
+  let notes: NoteTableView[] | null = [];
```

Comment:
Instead of mutating this it might be better to extract the logic into its own function?

Something like:
`const notes = await searchNotes(query, tag)`

Reply:
love it